packages/tasks/src/model-libraries.ts

```diff
@@ -1,52 +1,333 @@
+import * as snippets from "./library-ui-elements";
+import type { ModelData } from "./model-data";
+
+/**
+ * Elements configurable by a model library.
+ */
+export interface LibraryUiElement {
+	/**
+	 * Pretty name of the library.
+	 * displayed in tags, and on the main
+	 * call-to-action button on the model page.
+	 */
+	btnLabel: string;
+	/**
+	 * Repo name of the library's (usually on GitHub) code repo
+	 */
+	repoName: string;
+	/**
+	 * URL to library's (usually on GitHub) code repo
+	 */
+	repoUrl: string;
+	/**
+	 * URL to library's docs
+	 */
+	docsUrl?: string;
+	/**
+	 * Code snippet(s) displayed on model page
+	 */
+	snippets?: (model: ModelData) => string[];
+	/**
+	 * should we display this library in hf.co/models filter
+	 * (only for popular libraries with > 100 models)
+	 */
+	filter?: boolean;
+}
+
 /**
  * Add your new library here.
  *
  * This is for modeling (= architectures) libraries, not for file formats (like ONNX, etc).
- * File formats live in an enum inside the internal codebase.
+ * (unlike libraries, file formats live in an enum inside the internal codebase.)
+ *
+ * Doc on how to add a library to the Hub:
+ *
+ * https://huggingface.co/docs/hub/models-adding-libraries
+ *
+ * /!\ IMPORTANT
+ *
+ * The key you choose is the tag your models have in their library_name on the Hub.
  */
-export enum ModelLibrary {
-	"adapter-transformers" = "Adapters",
-	"allennlp" = "allenNLP",
-	"asteroid" = "Asteroid",
-	"bertopic" = "BERTopic",
-	"diffusers" = "Diffusers",
-	"doctr" = "docTR",
-	"espnet" = "ESPnet",
-	"fairseq" = "Fairseq",
-	"flair" = "Flair",
-	"keras" = "Keras",
-	"k2" = "K2",
-	"mlx" = "MLX",
-	"nemo" = "NeMo",
-	"open_clip" = "OpenCLIP",
-	"paddlenlp" = "PaddleNLP",
-	"peft" = "PEFT",
-	"pyannote-audio" = "pyannote.audio",
-	"sample-factory" = "Sample Factory",
-	"sentence-transformers" = "Sentence Transformers",
-	"setfit" = "SetFit",
-	"sklearn" = "Scikit-learn",
-	"spacy" = "spaCy",
-	"span-marker" = "SpanMarker",
-	"speechbrain" = "speechbrain",
-	"tensorflowtts" = "TensorFlowTTS",
-	"timm" = "Timm",
-	"fastai" = "fastai",
-	"transformers" = "Transformers",
-	"transformers.js" = "Transformers.js",
-	"stanza" = "Stanza",
-	"fasttext" = "fastText",
-	"stable-baselines3" = "Stable-Baselines3",
-	"ml-agents" = "Unity ML-Agents",
-	"pythae" = "Pythae",
-	"mindspore" = "MindSpore",
-	"unity-sentis" = "Unity Sentis",
-}
 
-export type ModelLibraryKey = keyof typeof ModelLibrary;
+export const MODEL_LIBRARIES_UI_ELEMENTS = {
+	"adapter-transformers": {
+		btnLabel: "Adapters",
+		repoName: "adapters",
+		repoUrl: "https://github.com/Adapter-Hub/adapters",
+		docsUrl: "https://huggingface.co/docs/hub/adapters",
+		snippets: snippets.adapters,
+		filter: true,
+	},
+	allennlp: {
+		btnLabel: "AllenNLP",
+		repoName: "AllenNLP",
+		repoUrl: "https://github.com/allenai/allennlp",
+		docsUrl: "https://huggingface.co/docs/hub/allennlp",
+		snippets: snippets.allennlp,
+		filter: true,
```

Comment:
Let's remove this one too as it's archived 

```suggestion
```

Reply:
even if it leads to 146 models?

<img width="864" alt="image" src="https://github.com/huggingface/huggingface.js/assets/326577/30febf28-e778-414a-ace1-ceb95475ea09">

Reply:
Yeah, with the library being archived I don't think it makes much sense anymore. It's 77 public models

Reply:
will do a more comprehensive cleanup in a later PR